File: 01-base-skill/Experiments/app/src/main/java/com/ztiany/test/AppContext.java
package com.ztiany.test;

import android.app.Application;
import android.content.Context;

import androidx.multidex.MultiDex;

import com.github.gzuliyujiang.oaid.DeviceID;

/**
 * @author Ztiany
 * Email: ztiany3@gmail.com
 * Date : 2020-06-16 16:37
 */
public class AppContext extends Application {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

}

File: 05-architecture/Kotlin-AndroidX/app/src/main/java/com/ztiany/kotlin/AppContext.kt
package com.ztiany.kotlin

import android.content.Context
import androidx.multidex.MultiDexApplication

/**
 *@author Ztiany
 *      Email: ztiany3@gmail.com
 *      Date : 2017-07-09 18:51
 */
class AppContext : MultiDexApplication() {

    private object C {
        lateinit var instance: AppContext
    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        C.instance = this
    }

    companion object {
        fun getInstance(): AppContext {
            return C.instance
        }
    }

}